src/CommunityToolkit.Mvvm.CodeFixers/AsyncVoidReturningRelayCommandMethodCodeFixer.cs

```diff
@@ -0,0 +1,93 @@
+
+    /// <inheritdoc/>
+    public override FixAllProvider? GetFixAllProvider()
+    {
+        return WellKnownFixAllProviders.BatchFixer;
+    }
+
+    /// <inheritdoc/>
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        Diagnostic diagnostic = context.Diagnostics[0];
+        TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+        // Retrieve the property passed by the analyzer
+        if (diagnostic.Properties[AsyncVoidReturningRelayCommandMethodAnalyzer.MethodNameKey] is not string methodName)
+        {
+            return;
+        }
+
+        SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+        // Get the method declaration from the target diagnostic
+        if (root!.FindNode(diagnosticSpan) is MethodDeclarationSyntax { Identifier.Text: string identifierName } methodDeclaration &&
+            identifierName == methodName)
+        {
+            // Register the code fix to update the return type to be Task instead
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: "Change return type to Task",
+                    createChangedDocument: token => ChangeReturnType(context.Document, root, methodDeclaration, token),
+                    equivalenceKey: "Change return type to Task"),
+                diagnostic);
+
+            return;
+        }
+    }
+
+    /// <summary>
+    /// Applies the code fix to a target method declaration and returns an updated document.
+    /// </summary>
+    /// <param name="document">The original document being fixed.</param>
+    /// <param name="root">The original tree root belonging to the current document.</param>
+    /// <param name="methodDeclaration">The <see cref="MethodDeclarationSyntax"/> to update.</param>
+    /// <param name="cancellationToken">The cancellation token for the operation.</param>
+    /// <returns>An updated document with the applied code fix, and the return type of the method being <see cref="Task"/>.</returns>
+    private static async Task<Document> ChangeReturnType(Document document, SyntaxNode root, MethodDeclarationSyntax methodDeclaration, CancellationToken cancellationToken)
+    {
+        // Get the semantic model (bail if it's not available)
+        if (await document.GetSemanticModelAsync(cancellationToken) is not SemanticModel semanticModel)
+        {
+            return document;
+        }
+
+        // Also bail if we can't resolve the Task symbol (this should really never happen)
+        if (semanticModel.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task") is not INamedTypeSymbol taskSymbol)
+        {
+            return document;
+        }
+
+        SyntaxNode typeSyntax = SyntaxGenerator.GetGenerator(document).TypeExpression(taskSymbol, addImport: true);
```

Comment:
This is what you need.

```suggestion
        SyntaxNode typeSyntax = SyntaxGenerator.GetGenerator(document).TypeExpression(taskSymbol).WithAdditionalAnnotations(Simplifier.AddImportsAnnotation);
```

Reply:
Ooooh, thank you! Wow that `TypeExpression` API is really not intuitive at all 🤣

Reply:
fwiw, `TypeExpression(symbol)` is equivalent to `TypeExpression(symbol, addImport: true)`.

For the import to be automatically added, two things are needed:

- The AddImportsAnnotation, which you were missing.
- A special annotation with kind `SymbolId` (see https://github.com/dotnet/roslyn/blob/63d40746d8899ab48f3f8452cd8459026c1f9d9a/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SymbolAnnotation.cs#L15-L18), which will be added automatically by SyntaxGenerator when using `TypeExpression(symbol)`